diff -Naur a/hypercard_dasm.cc b/hypercard_dasm.cc
--- a/hypercard_dasm.cc	2021-05-19 19:49:43.952529642 -0400
+++ b/hypercard_dasm.cc	2021-05-19 19:49:52.673319717 -0400
@@ -63,7 +63,7 @@
 
       // lowercase the line for pseudo-parsing
       string lowercase_line = line;
-      transform(lowercase_line.begin(), lowercase_line.end(), lowercase_line.begin(), tolower);
+      transform(lowercase_line.begin(), lowercase_line.end(), lowercase_line.begin(), ::tolower);
       size_t comment_start = lowercase_line.find("--");
       if (comment_start != string::npos) {
         lowercase_line.resize(comment_start);
