diff -Naur a/lib/as3/registry.h b/lib/as3/registry.h
--- a/lib/as3/registry.h	2021-09-19 10:50:13.193584976 -0400
+++ b/lib/as3/registry.h	2021-09-19 10:51:27.797837135 -0400
@@ -153,7 +153,6 @@
 asset_bundle_list_t*registry_getassets();
 
 // static multinames
-classinfo_t voidclass;
 classinfo_t* registry_getanytype();
 classinfo_t* registry_getarrayclass();
 classinfo_t* registry_getobjectclass();
diff -Naur a/lib/gfxpoly/poly.h b/lib/gfxpoly/poly.h
--- a/lib/gfxpoly/poly.h	2021-09-19 10:50:29.656199275 -0400
+++ b/lib/gfxpoly/poly.h	2021-09-19 10:51:51.638278626 -0400
@@ -18,7 +18,7 @@
     int32_t x;
     int32_t y;
 } point_t;
-type_t point_type;
+extern type_t point_type;
 
 #define SEGNR(s) ((int)((s)?(s)->nr:-1))
 
diff -Naur a/src/swfc-feedback.c b/src/swfc-feedback.c
--- a/src/swfc-feedback.c	2021-09-19 10:50:42.951887777 -0400
+++ b/src/swfc-feedback.c	2021-09-19 10:52:29.685387327 -0400
@@ -24,6 +24,11 @@
 #include <stdarg.h>
 #include "swfc-feedback.h"
 
+char* filename;
+int line;
+int column;
+void (*cleanUp)();
+
 void syntaxerror(char*format, ...)
 {
     char buf[1024];
diff -Naur a/src/swfc-feedback.h b/src/swfc-feedback.h
--- a/src/swfc-feedback.h	2021-09-19 10:50:51.464688337 -0400
+++ b/src/swfc-feedback.h	2021-09-19 10:52:56.435760688 -0400
@@ -22,10 +22,10 @@
 #ifndef __FEEDBACK_H
 #define __FEEDBACK_H
 
-char* filename;
-int line;
-int column;
-void (*cleanUp)();
+extern char* filename;
+extern int line;
+extern int column;
+extern void (*cleanUp)();
 
 void syntaxerror(char*format, ...);
 void warning(char*format, ...);
diff -Naur a/src/swfc-history.c b/src/swfc-history.c
--- a/src/swfc-history.c	2021-09-19 10:51:00.743470953 -0400
+++ b/src/swfc-history.c	2021-09-19 10:53:17.124276061 -0400
@@ -23,6 +23,12 @@
 #include <memory.h>
 #include "swfc-history.h"
 
+FILTER* noFilters;
+FILTER_BLUR* noBlur;
+FILTER_BEVEL* noBevel;
+FILTER_DROPSHADOW* noDropshadow;
+FILTER_GRADIENTGLOW* noGradientGlow;
+
 enum
 {
     T_BEFORE,
diff -Naur a/src/swfc-history.h b/src/swfc-history.h
--- a/src/swfc-history.h	2021-09-19 10:51:08.647285784 -0400
+++ b/src/swfc-history.h	2021-09-19 10:53:31.107948501 -0400
@@ -55,11 +55,11 @@
 
 #define IF_FIXED_ALIGNMENT 0x0001
 
-FILTER* noFilters;
-FILTER_BLUR* noBlur;
-FILTER_BEVEL* noBevel;
-FILTER_DROPSHADOW* noDropshadow;
-FILTER_GRADIENTGLOW* noGradientGlow;
+extern FILTER* noFilters;
+extern FILTER_BLUR* noBlur;
+extern FILTER_BEVEL* noBevel;
+extern FILTER_DROPSHADOW* noDropshadow;
+extern FILTER_GRADIENTGLOW* noGradientGlow;
 
 typedef struct _spline
 {
