diff -Naur a/lib/as3/registry.h b/lib/as3/registry.h
--- a/lib/as3/registry.h	2021-09-19 10:46:45.341455336 -0400
+++ b/lib/as3/registry.h	2021-09-19 10:46:54.557239368 -0400
@@ -153,7 +153,6 @@
 asset_bundle_list_t*registry_getassets();
 
 // static multinames
-classinfo_t voidclass;
 classinfo_t* registry_getanytype();
 classinfo_t* registry_getarrayclass();
 classinfo_t* registry_getobjectclass();
