diff -Naur a/src/abydos.c b/src/abydos.c
--- a/src/abydos.c	2020-05-20 09:12:43.305008580 -0400
+++ b/src/abydos.c	2020-05-20 09:13:30.031928841 -0400
@@ -774,6 +774,16 @@
 
 /* --- info ------------------------------------------------------------- */
 
+int abydos_list_mimetypes_itertor(void * key, void * value, void * userdata)
+{
+	printf("item\n");
+}
+
+void abydos_list_mimetypes()
+{
+	hash_map_foreach(plugin_map, abydos_list_mimetypes_itertor, 0);
+}
+
 void
 abydos_get_info_V(abydos_t *ar,abydos_info_t *info,int info_version)
 {
diff -Naur a/src/abydos.h b/src/abydos.h
--- a/src/abydos.h	2020-05-20 09:12:53.608990980 -0400
+++ b/src/abydos.h	2020-05-20 09:13:11.599960272 -0400
@@ -138,6 +138,7 @@
 int abydos_load_feed(abydos_t *ar, const char* data, size_t len);
 int abydos_load_end(abydos_t *ar);
 
+void abydos_list_mimetypes();
 void abydos_get_info_V(abydos_t *ar,abydos_info_t *info,int);
 #define abydos_get_info(a,b) abydos_get_info_V(a,b,ABYDOS_INFO_VERSION)
 void abydos_get_size(abydos_t *ar,int *width,int *height,int flags);
