diff '--color=auto' -Naur a/src/6502emu.c b/src/6502emu.c
--- a/src/6502emu.c	2025-06-15 17:06:33.025657857 -0400
+++ b/src/6502emu.c	2025-06-15 17:06:51.192849465 -0400
@@ -205,14 +205,14 @@
     r->flags |= (value == 0 ? FLAG_Z : 0) | (value & FLAG_N);
 }
 
-static void update_carry(struct cpu_ctx *r, int bool)
+static void update_carry(struct cpu_ctx *r, int b)
 {
-    r->flags = (r->flags & ~FLAG_C) | (bool != 0 ? FLAG_C : 0);
+    r->flags = (r->flags & ~FLAG_C) | (b != 0 ? FLAG_C : 0);
 }
 
-static void update_overflow(struct cpu_ctx *r, int bool)
+static void update_overflow(struct cpu_ctx *r, int b)
 {
-    r->flags = (r->flags & ~FLAG_V) | (bool != 0 ? FLAG_V : 0);
+    r->flags = (r->flags & ~FLAG_V) | (b != 0 ? FLAG_V : 0);
 }
 
 static void op_adc(struct cpu_ctx *r, int mode, union inst_arg *arg)
