diff -Naur a/scfbuild/builder.py b/scfbuild/builder.py
--- a/scfbuild/builder.py	2020-08-21 14:09:29.940657345 -0400
+++ b/scfbuild/builder.py	2020-08-21 14:10:09.851601736 -0400
@@ -288,7 +288,7 @@
         # table__n_a_m_e.setName().
         record = NameRecord()
         # PyYAML creates strings, force to Unicode
-        record.string = unicode(text)
+        record.string = str(text)
         record.nameID = name_id
         record.platformID = platform_id
         record.platEncID = plat_enc_id
