diff -Naur a/dlls/kernel32/heap.c b/dlls/kernel32/heap.c
Patch from: https://bugs.winehq.org/show_bug.cgi?id=44931
--- a/dlls/kernel32/heap.c	2021-01-05 09:38:46.252855553 -0500
+++ b/dlls/kernel32/heap.c	2021-01-05 09:40:13.660711641 -0500
@@ -597,11 +597,14 @@
          lpBuffer->dwAvailPageFile = 2U*1024*1024*1024 -  lpBuffer->dwAvailPhys - 1;
 
     /* limit page file size for really old binaries */
+    /* use MAXLONG/2, so that dwAvailPhys + dwAvailPageFile < MAXLONG */
     if (nt->OptionalHeader.MajorSubsystemVersion < 4 ||
         nt->OptionalHeader.MajorOperatingSystemVersion < 4)
     {
-        if (lpBuffer->dwTotalPageFile > MAXLONG) lpBuffer->dwTotalPageFile = MAXLONG;
-        if (lpBuffer->dwAvailPageFile > MAXLONG) lpBuffer->dwAvailPageFile = MAXLONG;
+    	lpBuffer->dwTotalPhys = min(lpBuffer->dwTotalPhys, MAXLONG / 2);
+    	lpBuffer->dwAvailPhys = min(lpBuffer->dwAvailPhys, MAXLONG / 2);
+    	lpBuffer->dwTotalPageFile = min(lpBuffer->dwTotalPageFile, MAXLONG / 2);
+    	lpBuffer->dwAvailPageFile = min(lpBuffer->dwAvailPageFile, MAXLONG / 2);
     }
 
     TRACE("Length %u, MemoryLoad %u, TotalPhys %lx, AvailPhys %lx,"
