diff -Naur a/helpdec1.c b/helpdec1.c
--- a/helpdec1.c	2021-05-23 21:27:26.545003471 -0400
+++ b/helpdec1.c	2021-05-23 21:27:40.527667705 -0400
@@ -29,8 +29,8 @@
     va_start(arg,format);
     vfprintf(stderr,format,arg);
     va_end(arg);
-    fputs("\nPress CR to continue at your own risk, any other key to exit.\n",stderr);
-    if(getch()!='\r') exit(1);
+    /*fputs("\nPress CR to continue at your own risk, any other key to exit.\n",stderr);
+    if(getch()!='\r') exit(1);*/
 }
 
 #ifndef HAVE_STRNCPY
