diff '--color=auto' -Naur a/build.gradle b/build.gradle
--- a/build.gradle	2024-10-31 11:10:20.636508425 -0400
+++ b/build.gradle	2024-10-31 11:10:11.634358344 -0400
@@ -7,8 +7,8 @@
 group = 'net.arnx'
 version = '0.9.11'
 
-sourceCompatibility = 1.6
-targetCompatibility = 1.6
+sourceCompatibility = 8
+targetCompatibility = 8
 
 repositories {
     mavenCentral()
@@ -59,12 +59,12 @@
 }
 
 task sourcesJar(type: Jar) {
-    classifier 'sources'
+    archiveClassifier.set('sources')
     from sourceSets.main.allSource
 }
 
 task javadocJar(type: Jar, dependsOn: javadoc) {
-    classifier 'javadoc'
+    archiveClassifier.set('javadoc')
     from javadoc.destinationDir
 }
 
@@ -119,4 +119,4 @@
 
 signing {
 	sign publishing.publications.mavenJava
-}
\ No newline at end of file
+}
