diff -Naur a/libs/powzix/kraken.cpp b/libs/powzix/kraken.cpp
--- a/libs/powzix/kraken.cpp	2021-06-07 16:03:48.936907294 -0400
+++ b/libs/powzix/kraken.cpp	2021-06-07 16:04:15.776882735 -0400
@@ -19,13 +19,13 @@
 
 #include "stdafx.h"
 
- static unsigned int _rotl(  unsigned int value, int shift )
+/* static unsigned int _rotl(  unsigned int value, int shift )
  {
      int max_bits = sizeof(value)<<3; 
      if ( shift > max_bits )
          shift = shift % max_bits;
      return (value << shift) | (value >> (max_bits-shift));
- }
+ }*/
 
 // Header in front of each 256k block
 typedef struct KrakenHeader {
